db2_test.go

```diff
@@ -323,3 +326,21 @@ func TestPushValueLogLimit(t *testing.T) {
 		}
 	})
 }
+
+// The following benchmark test is supposed to be run against a badger directory with some data.
+// Use badger fill to create data if it doesn't exist.
+func BenchmarkDBOpen(b *testing.B) {
+	if *benchDir == "" {
+		b.Skip("Please set -benchdir to badger directory")
+	}
+	dir := *benchDir
+	opt := DefaultOptions
+	opt.Dir = dir
+	opt.ValueDir = dir
+	opt.ReadOnly = true
```

Comment:
Will benchmark produce different results if `ReadOnly` is not set?

Reply:
Yes. It would. In read only mode we do not perform compactions. Once the DB is completely loaded, compactions might automatically trigger and affect the benchmark.